Makefile

```diff
@@ -351,4 +351,16 @@ test:
 	skipper make $(SKIPPER_PARAMS) _test
 
 _test: $(REPORTS)
+	if [ -d /tmp/assisted_test_infra_logs ]; then rm -rf /tmp/assisted_test_infra_logs; fi
+	mkdir /tmp/assisted_test_infra_logs
+	cp -p discovery-infra/test_infra/tools/tf_network_pool.json /tmp/tf_network_pool.json
 	python3 -m pytest $(or ${TEST},discovery-infra/tests) -k $(or ${TEST_FUNC},'') -m $(or ${TEST_MARKER},'') --verbose -s --junit-xml=$(REPORTS)/unittest.xml
+
+test_parallel:
+	if [ -d /tmp/assisted_test_infra_logs ]; then rm -rf /tmp/assisted_test_infra_logs; fi
+	mkdir /tmp/assisted_test_infra_logs
+	cp -p discovery-infra/test_infra/tools/tf_network_pool.json /tmp/tf_network_pool.json
+	skipper make $(SKIPPER_PARAMS) _test_parallel
+
+_test_parallel: $(REPORTS)
+	python3 -m pytest -n $(or ${MULTI_NUM}, '2') $(or ${TEST},discovery-infra/tests) -k $(or ${TEST_FUNC},'') -m $(or ${TEST_MARKER},'') --verbose -s --junit-xml=$(REPORTS)/unittest.xml
```

Comment:
tf_network_pool.json only has 2 networks. what happens if MUTLI_NUM > 2? The number of networks in itf_network_pool.json should match the number of parallel tests. Maybe remove this option until this is supported?

Reply:
@lalon4 lets add logic that if we don't have free asset we just wait for it

Reply:
@YuviGold regarding merging , -n 1 != without -n. 
If u will want to debug your tests better to run without -n cause adding -n option sort of disables the interactive option.
The problem is that this plugin runs in subprocess. Better to have 2 those options or we can have one option that will validate if parallel == 1 not to add -n

Reply:
@tsorya its a bit tricky cause its inside the lock, so waiting will keep the file locked.
Cant we just declare that we are supporting up to X number of tests in parallel?
for now i will add 3 or 4 (depends on the time i have to test it).

Reply:
i think better to have wait

Reply:
I also think it is nice to have, and not hard to implement. but i think we should not implement such a mechanism.
is_free_asset should be done outside the Asset obj, therefore there will be X processes that will hang and wait for free asset. meanwhile, each process should query a file (load into memo) in a loop, and in some cases for hours (until free asset). 
while we have 15 VM running, containers, and many more processes.
So i think this mechanism will consume more resources. I also have no time to invest in testing such cases. 
Now we have 40 tests, next month 100... dont know the influence 
In this PR scope, added 3 assets and this is what was tested. So for start we will support 3
WDYT?